Classes/Plugin/Collection.php

```diff
@@ -129,12 +129,10 @@ protected function showCollectionList() {
                 ),
                 Helper::whereExpression('tx_dlf_collections')
             )
-            ->orderBy($orderBy)
-            ->execute();
-
-        $allResults = $result->fetchAll();
+            ->orderBy($orderBy);
 
-        $count = count($allResults);
+        $result = $queryBuilder->execute();
+        $count = $queryBuilder->count('uid')->execute()->fetchColumn(0);
```

Comment:
Does this really count the number of results for the query above? It seems the count the total number of collections in the database.

Reply:
Yes it does work. It is the same prepared query in  $queryBuilder, added the COUNT('uid') function.